### PR TITLE
Create-sqrt-monotonic-binary-search.cpp

### PR DESCRIPTION
**Pull request by hazel11182**

### CODE REVIEW

**Comment by TawfikYasser** [comment]
Please explain what is this code do? @hazel11182

**Comment by hazel11182** [comment]
we are finding the squareroot of any given given number  till 3 decimals using binary search. and we applied the concept of binary search in monotonic functions.

**Comment by hazel11182** [comment]
@ming-tsai now is it fine?

**Comment by ming-tsai** [comment]
Hi @Arsenic-ATG,
Could help me on this pull request?

**Comment by ming-tsai** [comment]
> @ming-tsai now is it fine?

Could you explain me what will be the difference between binary search on monotonic function and normal binary search?

**Comment by Arsenic-ATG** [comment]
> Could you explain me what will be the difference between binary search on monotonic function and normal binary search?

@ming-tsai there is no difference in this case. 

it is one of the used cases where binary search will be a viable option to choose as linear search will take tremendous amount of time to do the same task.

although a program is already there implanting a classical binary search on a list or array of discrete items, but this is also a nice used case where binary search is being used on continuous set of values ( example where `target_value` belongs to closed interval [0,n] ) instead of discrete set of values ( for example where `target_value` belongs to { 1 , 4, 5 ,12 , 15 } ).

**Review by Arsenic-ATG — CHANGES_REQUESTED** [review]
the program functionally works ✅
 
but you forgot some minor things :- 

-  updating the [index readme file](https://github.com/MakeContributions/DSA/blob/main/searching/README.md) with your creation in the list 

- prefer using `-` as word separator instead of under-sore(`_`)

all of this is already mentioned in the [repository's contribution guidelines](https://github.com/MakeContributions/DSA#1-contribution-specifications) which is where you should go for further details about the same.

**Comment by hazel11182** [comment]
@Arsenic-ATG what needs to be done now?

**Comment by Arsenic-ATG** [comment]
> @Arsenic-ATG what needs to be done now?

just perform the required changes on your forked branch (https://github.com/hazel11182/DSA/tree/patch-3), the changes would be automatically reflected back to this pull request.

**Comment by hazel11182** [comment]
done

File: searching/c-or-cpp/sqrt_monotonic_binary_search.cpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#include <iostream>
```
Comment (ming-tsai) [inline searching/c-or-cpp/sqrt_monotonic_binary_search.cpp:1]: Hi @hazel11182,
Could replace the file name from **_** to **-**?

**Comment by Arsenic-ATG** [comment]
> done

@hazel11182 I think you got it wrong, we are talking about the name of the file you created ( [sqrt_monotonic_binary_search.cpp](https://github.com/hazel11182/DSA/blob/patch-3/searching/c-or-cpp/sqrt_monotonic_binary_search.cpp) ) and not the title of the pull request. 

and [this](https://github.com/hazel11182/DSA/blob/patch-3/searching/README.md) is your index file that you have to add your program to at the end of the list under `c  or cpp` section with a relative link to your code file. 

I have done the required changes for you in the [other pull request](https://github.com/MakeContributions/DSA/pull/113) and you make take reference from that about how this should be done.


-----

P.S. 
if you don't know how to make a text a hyperlink then use the following syntax 
```
[text](https://github.com/MakeContributions/DSA/pull/114)
```
which will look like this 

[text](https://github.com/MakeContributions/DSA/pull/114)


**you may look at [this](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-readmes#relative-links-and-image-paths-in-readme-files) guide for more information about creating relative links in any markdown file.**

----

**Comment by Arsenic-ATG** [comment]
###  🚧 closing this PR because of long inactivity. 

If someone want to work on this then feel free to reopen it or create a new PR with the required changes.

**Comment by hazel11182** [comment]
How do i reopen PR?

**Comment by hazel11182** [comment]
HI, sorry for being inactive. I have resolved the file name. Is it nowfine?

**Review by Arsenic-ATG — DISMISSED** [review]

**Comment by Arsenic-ATG** [comment]
This PR will conflict with Pr #127 as both upgrade readme files.

**Comment by ming-tsai** [comment]
I will create merge with both pr

**Review by ming-tsai — APPROVED** [review]

**Pull request merged**